src/main/java/slimeknights/tconstruct/library/modifiers/modules/technical/MaxArmorStatModule.java

```diff
@@ -0,0 +1,152 @@
+    return tooltipStyle == ArmorStatModule.TooltipStyle.NONE ? NO_TOOLTIP_HOOKS : TOOLTIP_HOOKS;
+  }
+
+  @Override
+  public void onEquip(IToolStackView tool, ModifierEntry modifier, EquipmentChangeContext context) {
+    if (condition.matches(tool, modifier)) {
+      context.getTinkerData().ifPresent(data -> {
+        addStatIfArmor(tool, context, statKey, maxLevel, amount, modifier, allowBroken, heldTag);
+      });
+
+    }
+  }
+
+  @Override
+  public void onUnequip(IToolStackView tool, ModifierEntry modifier, EquipmentChangeContext context) {
+    if (condition.matches(tool, modifier)) {
+      context.getTinkerData().ifPresent(data -> {
+        addStatIfArmor(tool, context, statKey, maxLevel, amount, modifier, allowBroken, heldTag);
+      });
+    }
+  }
+
+  @Override
+  public void addTooltip(IToolStackView tool, ModifierEntry modifier, @Nullable Player player, List<Component> tooltip, TooltipKey tooltipKey, TooltipFlag tooltipFlag) {
+    if (condition.matches(tool, modifier) && (tool.hasTag(TinkerTags.Items.WORN_ARMOR) || heldTag != null && tool.hasTag(heldTag)) && (!tool.isBroken() || allowBroken)) {
+      float value = player != null ? ArmorStatModule.getStat(player, statKey) : modifier.getEffectiveLevel();
+      if (value != 0) {
+        Component name = Component.translatable(Util.makeTranslationKey("armor_stat", statKey.getId()));
+        switch (tooltipStyle) {
+          case BOOST -> TooltipModifierHook.addFlatBoost(modifier.getModifier(), name, value, tooltip);
+          case PERCENT -> TooltipModifierHook.addPercentBoost(modifier.getModifier(), name, value, tooltip);
+        }
+      }
+    }
+  }
+
+  /**
+   * Adds to the armor stat for the given maxLevel if the tool is in a valid armor slot
+   * @param tool          Tool instance
+   * @param context       Equipment change context
+   * @param statKey       Key to modify
+   * @param maxLevelKey   Key for max level of armor modifier
+   * @param amount        Amount by which stat is modified per level
+   * @param modifierEntry Modifier entry for stat
+   * @param allowBroken   Whether the tool can work while broken
+   * @param heldTag       Tag to check to validate held items, null means held disallowed
+   */
+  public static void addStatIfArmor(IToolStackView tool, EquipmentChangeContext context, TinkerDataCapability.TinkerDataKey<Float> statKey, TinkerDataCapability.ComputableDataKey<ModifierMaxLevel> maxLevelKey, LevelingValue amount, ModifierEntry modifierEntry, boolean allowBroken, @Nullable TagKey<Item> heldTag) {
+    if (ArmorLevelModule.validSlot(tool, context.getChangedSlot(), heldTag) && (!tool.isBroken() || allowBroken)) {
+      context.getTinkerData().ifPresent(data -> {
+        float oldLevel = data.computeIfAbsent(maxLevelKey).getMax();
+        float newLevel = MaxArmorLevelModule.updateMaxLevel(context, maxLevelKey, modifierEntry);
+        data.add(statKey, amount.compute(newLevel - oldLevel));
```

Comment:
I'm also noticing a subtle bug here, `LevelingAmount` has both a constant part and a leveling part. The constant part is going to get added every time this is called with the current code. I think the best option would be to modify `LevelingValue#compute` to return `0` if the level is `0`, though I don't feel like analyzing the risk of that. Instead lets add a new helper to `LevelingValue` called `computeForLevel`. We need to call this new helper one for the old level and once for the new, we cannot simply call it on the difference:

```java
public float computeForLevel(float level) {
  return level > 0 ? compute(level) : 0;
}
```

Also, bit of a nit pick, but would rather just fetch the data key once, the helper is not really helping providing any benefit as it has to do a lot of extra work (notably with the cap resolve). Might as well just call set here directly. Updated code would look like:

```java
ModifierMaxLevel maxLevel = data.computeIfAbsent(maxLevelKey);
float oldLevel = maxLevel.getMax();
maxLevel.set(context.getChangedSlot(), modifierEntry.getEffectiveLevel());
data.add(statKey, amount.computeForLevel(maxLevel.get()) - amount.computeForLevel(oldLevel));
```